include/deal.II/base/parameter_acceptor.h

```diff
@@ -516,6 +520,72 @@ class ParameterAcceptor : public Subscriptor
 
 
 
+/**
+ * A Proxy ParameterAcceptor wrapper for classes that have a static member
+ * function declare_parameters, and a non virtual parse_parameters method.
+ *
+ * If a class already implements the static member function declare_parameters,
+ * and the method declare_parameters, then it cannot be derived directly from
+ * ParameterAcceptor, as the existing member functions would be incompatible
+ * with the virtual methods with the same name that ParameterAcceptor expects
+ * you to overload. For these cases, you can still use ParameterAcceptor
+ * facilities and logic by wrapping your class inside this proxy. The resulting
+ * class will have the public interface of your SourceClass, and will provide
+ * the correct interface to ParameterAcceptor.
+ *
+ * An example usage is given by the following snippet of code, using
+ * Functions::ParsedFunction as an example source class:
+ *
+ * @code
+ * ParameterAcceptorProxy<Functions::ParsedFunction<2> > fun("Some function");
+ * ParameterAcceptor::initialize("test.prm");
+ * @endcode
+ *
+ * The above snippet of code will initialize ParameterAcceptor::prm with a
+ * section "Some function", and will correctly parse and assing to the object
```

Comment:
How does this work if one has an existing parameter file with a single section missing? Does this overwrite the entire file or just append the missing section to it?

Reply:
`ParameterAcceptor` just calls `declare_parameters` and `parse_parameters` for each of its subclass. This does not stop you from having other parameters, as long as you declare them first. Nothing is done to the existing ones.

Reply:
assing -> assign